src/lib/core.d.ts

```diff
@@ -1011,12 +1011,12 @@ interface Array<T> {
       * Combines two or more arrays.
       * @param items Additional items to add to the end of array1.
       */
-    concat<U extends T[]>(...items: U[]): T[];
+    concat<U extends T[]>(...items: (U | U[])[]): T[];

```

Comment:
I don't actually know why this was generic in the first place. Is there a reason `T[]` couldn't suffice?

Additionally, even if we decide not to drop this overload, this change is incorrect, since `[1,2,3].concat([[1]],[[23]])` now works and still returns `number[]`. Give the following a try in the playground:

```ts
interface Array<T> {
    myConcat<U extends T[]>(...items: (U | U[])[]): T[];
    myConcat(...items: (T | T[])[]): T[];
}

var x = [1,2,3].myConcat([[23]])
```